api/controller/dbController.js

```diff
@@ -0,0 +1,86 @@
+import { query } from '../../db/db'
+import { 
+  artistQueries, setlistQueries, songQueries, setlistSongQueries
+  } from '../../db/queries'
+
+  export class dbController {
+    
+    addAllInfo = async ({apiResult,externalId,artistName}) => {
+      const artistId = await this.addedArtist({externalId,artistName})
+      let listOfShows = apiResult.setlist
+      for (const concert of listOfShows){
+        let concertId = await this.addSetlist({artistId, concertExternalId:concert.id, updatedAt:concert.lastUpdated, eventDate:concert.eventDate})
+        let concertSets = concert.sets.set
+        if (concertSets.length > 0){
+          let songsFromThisConcert = this.deriveSongsFromSets(concertSets);
+          for (const songName of songsFromThisConcert){
+            let returnedSong = await this.getSong({songName,artistId})
+            let songId = !returnedSong ? await this.addSong({songName,artistId}): returnedSong;
+            await this.addSetlistSong({concertId,songId});
+          }
+        }
+      }
+      return artistId
+    }
+
+    getDates = async ({artistId,songName}) => {
+      const returnedValue = {};
+      const result = await query(setlistSongQueries.getSetlistSong,[songName,artistId])
+      const compiledDates = this.datesFromDateObject(result);
+      returnedValue[songName] = compiledDates;
+      return returnedValue;
+    }
+
+    addSong = async ({songName,artistId}) => {
+      const addResult = await query(songQueries.addSong,[songName,artistId]);
+      return addResult[0].id;
+    }
+    
+    getSong = async ({songName,artistId}) => {
+      const getResult = await query(songQueries.getSong,[songName,artistId]);
+      const songInDb = getResult.length > 0;
+      if (songInDb) {
+        return getResult[0].id;
+      } else {
+        return null;
+      }
+    }
+
+    addSetlistSong = async ({concertId,songId}) => {
+      await query(setlistSongQueries.addSetlistSong,[concertId,songId])
+    }
+
+    addSetlist = async ({artistId,concertExternalId,updatedAt,eventDate}) => {
+      const result = await query(setlistQueries.addSetlist,[concertExternalId,artistId,updatedAt,eventDate]);
+      return result[0].id;
+    }
+  
+    addedArtist = async ({externalId,artistName}) => {
+      const result = await query(artistQueries.addArtist,[externalId,artistName]);
+      return result[0].id;
+    }
+  
+    getArtist = async (artistName) => {
+      console.log(artistName)
+      const result = await query(artistQueries.getArtist,[artistName]);
+      return result.length > [0] ? result[0].id : 0;
```

Comment:
Not sure how `result.length > [0]` is working... I guess it's always returning true? Should be `> 0`. There might be ways to more elegantly put this code. I think this would work?
```
return result[0] ? result[0].id : 0
```
Or if we don't care about `0` and are fine with `undefined` then we can do this `return result[0]?.id` If you haven't used optional chaining yet, it's pretty cool, [you can check it out here.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)